proto/src/determined/trial/v1/trial.proto

```diff
@@ -150,3 +150,30 @@ message TrialProfilerMetricsBatch {
   // The labels for this series.
   TrialProfilerMetricLabels labels = 4;
 }
+
+// Signals to the experiment the trial early exited.
+message TrialEarlyExit {
+  // The reason for an early exit.
+  enum ExitedReason {
+    // Zero-value (not allowed).
+    EXITED_REASON_UNSPECIFIED = 0;
+    // Indicates the trial exited due to an invalid hyperparameter.
+    EXITED_REASON_INVALID_HP = 1;
+    // Indicates the trial exited due to a user requested stop.
+    EXITED_REASON_USER_REQUESTED_STOP = 2;
+  }
+  // The reason for the exit.
+  ExitedReason reason = 1;
+}
+
+// The training metrics for some duration of training.
+message TrainingMetrics {
+  // The number of batches trained on when these metrics were reported.
+  int32 batches_trained = 1;
```

Comment:
q: are those two fields both required? If they are not set, what happens?

Reply:
i believe batches_trained (as a total amount) is required. @rb-determined-ai had more opinions about the python side after what is required/what is not.